### ISSUE TITLE
Chapter 7 : Gradient Boosting Algorithm : loss function is incorrect

### ISSUE THREAD

**Issue by pschhikara88** [rasbt/machine-learning-book#239]
<img width="879" height="696" alt="Image" src="https://github.com/user-attachments/assets/cd4768a5-01c4-4cc9-9993-f35290c3b16c" />

Formula should be : 

Incorrect formula 
`Li = - yi log(Pi) + (1- yi)log(1-Pi)

Correct formula 
```
Either
Li = -  ( yi log(Pi) + (1- yi)log(1-Pi) )
or

Li = - yi log(Pi) - (1- yi)log(1-Pi)

```

Because i tried to derive the above incorrect formula to 
` Li = log(1+ e^yi) - yi*yihat`
and then i realised, initial loss function is not correct. 

Please let me know if my understanding is correct.

**Comment by aditya12343212** [comment]
Hey! Just wanted to point out a small but important sign error in the binary cross-entropy loss formula that appears in some parts of the code/docs.

### Incorrect version (missing the negative sign on the second term):
```math
L_i = -y_i \log(p_i) + (1 - y_i) \log(1 - p_i)

**Comment by rasbt** [comment]
Sorry, took me a while to catch up. Yes, you are right. Adding this to the errata and will fix it in the next edition. I am closing the comment to keep track of the issues that I already incorporated. Thanks again for reporting this.
